src/Cli/dotnet/Commands/Test/TestingPlatformCommand.cs

```diff
@@ -27,53 +27,58 @@ public TestingPlatformCommand(string name, string description = null) : base(nam
 
     public int Run(ParseResult parseResult)
     {
-        int exitCode = ExitCode.Success;
+        int? exitCode = null;
         try
         {
-            ValidationUtility.ValidateMutuallyExclusiveOptions(parseResult);
+            exitCode = RunInternal(parseResult);
+        }
+        finally
+        {
+            CompleteRun(exitCode);
+            CleanUp();
+        }
 
-            PrepareEnvironment(parseResult, out TestOptions testOptions, out int degreeOfParallelism);
+        return exitCode ?? ExitCode.GenericFailure;
```

Comment:
`exitCode` should never be null here, I think.

Reply:
According to the compiler it can. I can more the return to the try, but then I need to make sure I assign the exitcode value before leaving the try to make it available to finally, and that is even less obvious.

Reply:
@nohwnd That sounds strange to me. Maybe it *was* when you had a `catch`?

Reply:
the code above will have to do:

exitCode = RunInternal(parseResult);
return exitCode;

Because in the success case we need exitCode assigned so finally can consume it.

if we throw, we want exitcode to remain null so we don't write the "testing platform exited with this testing platform exit code" .

seems easy to not assign the exit code and just do `return RunInternal` if you don't pay attention (I did it myself ofc). So I prefer the code above where we return at the end .